ironfish-rust/src/public_account/description.rs

```diff
@@ -0,0 +1,275 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use ed25519_dalek::{Signature, Verifier, VerifyingKey};
+use std::io;
+
+use crate::{
+    errors::{IronfishError, IronfishErrorKind},
+    PublicAddress,
+};
+
+use super::transfer::{Transfer};
+
+#[derive(Clone)]
+pub struct PublicAccountDescription {
+    /// TODO(jwp): do we need a separate account address for later migration
+    /// pub(crate) account_address: EthereumAddress
+    version: u8,
+    // Minimum number of signers required to sign a message and be valid
+    min_signers: u16,
+    // Signers of the public account
+    signers: Vec<VerifyingKey>,
+    // Signatures of the signers for a given message
+    signatures: Vec<Signature>,
+    // asset transfers
+    transfers: Vec<Transfer>,
+    // address of the account
+    address: PublicAddress,
+}
+
+impl PublicAccountDescription {
+    pub fn new(
+        version: u8,
+        min_signers: u16,
+        signers: Vec<VerifyingKey>,
+        transfers: Vec<Transfer>,
+        address: PublicAddress,
+    ) -> Result<PublicAccountDescription, IronfishError> {
+        let description = Self {
+            version,
+            min_signers,
+            signers,
+            signatures: vec![],
```

Comment:
No signatures when instantiating (need to do this if we want to be able to use `hash` with `&self`)